### ISSUE TITLE
[ingress-nginx] added exposing-tcp-udp-services

### ISSUE THREAD

**Issue by alexey-gavrilov-flant** [deckhouse/deckhouse#2149]
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

sometimes for an application, in addition to http, you also need to provide a specific port on the same IP as http

### Proposed Solution

add the ability to add ports

### Additional Information

https://kubernetes.github.io/ingress-nginx/user-guide/exposing-tcp-udp-services/

**Comment by EvgenySamoylov** [comment]
This is wrong approach. Need to use one of the recommended ways instead: https://kubernetes.io/docs/tutorials/kubernetes-basics/expose/expose-intro/

**Comment by nabokihms** [comment]
There is a good case that because of security reasons, it is better to proxy everything through a single hole, and this hole can be an ingress nginx. However, we should think about the implementation. Do we want to allow proxying both HTTP and tcp/udp for a single ingress or not.
